### DISCUSSION TITLE
Report in xls format show strange bold lines in empty cells after upgrading to v4.16

### DISCUSSION THREAD

**Discussion by PenghaiZhang** [eclipse-birt/birt#1866] [General]
After upgrading BIRT from v.4.9 to v.4.16, the report generated in xls format has some strange diagonal lines in empty cells. Please see the attached image (v4.16 on left hand side and v4.9 on right hand side). Our APP uses BIRT APIs to generate the report, and we have not changed any code. Is this a known bug or do we miss any new configuration for xls ? 

![image](https://github.com/user-attachments/assets/348d2903-6b7f-4f9c-8954-ee59bc31c601)

**Comment by speckyspooky** [comment]
The bold lines seems to be diagonal lines which are available since BIRT 4.14
I don't know what the reason is that the lines will be displayed directly on your report.

Can you please create the report like XLSX instead of XLS to see if there a difference
because the master implementation is done for the XLSX (XLS goes more and more out of scope due to the age of this file format).

Could you please provide a small example without external dependencies with this diagonals.
So we would be able to catch the reason of it.

**Reply by speckyspooky** [reply]
Ok, I have seen a hint on your video of the word topic
and it seems to me that you use the original excel-emitter "xls" instead of "spudsoft_xls" or "xlsx" - see my screen from your discussion #1818.

The difference is that the "xls"-emitter was only a prototype and yes, if I start the Excel-creation with the old emitter I get the diagonals too.

Please switch to the spudsoft_xls  or xlsx because the old excle-emitter of xls won't be supported any more.
And the old prototype-excel produce not a real xls it is a excel-xml with the extention "xls"
but with a complete different technical specification. So you will get a message from excel that the file extention is unequal to the content. 

![grafik](https://github.com/user-attachments/assets/16058f20-a11f-412d-8d45-6dc2944bf51c)

**Reply by PenghaiZhang** [reply]
Hi @speckyspooky , thanks a lot for your suggestion! I changed the emitter to `xlsx` and `xls_spudsoft`, and both will remove the diagonal lines. 

However, it seems the content generated by these two emitters are not quite right.

![image](https://github.com/user-attachments/assets/d784f602-c672-490e-98c6-5c0281c485a8)

In above screenshot, where the left one was generated by the `xls_spudsoft`, you can see some cells were merged into one cell, but in the right one which was generated by the original emiiter, the contents look right (except the diagonal lines).

Can you please use the attached report file and see if you can reproduce the issue ? 

Appreciate your investigation and help!

[commercial_table_with_no_dataset.zip](https://github.com/user-attachments/files/16702748/commercial_table_with_no_dataset.zip)

**Comment by speckyspooky** [comment]
I see it and the reason is very simple.
The spudsoft emitter (xls, xlsx) have a different logic to the excel-prototpye.

The spudsoft-emitter create the table based on the table/grid on top level of the report which has the most columns.
In your case you hase 1 grid at top level with 1 column and all other tables and grids are inserted tables/grids on a sub level.

So I changed your (data-)-table to the 1st table at top level and added all necessary rows to the header.
I removed also your total grid and added 2 columns for the first content column which is megred for the data, but regular splitted for your total.
The result is a perfect table. If you change the column width you will get a smaller table.
Attached you will get a screen of my result and also my test report and the excel-xlsx-result of it.

### Screen of the new result
![grafik](https://github.com/user-attachments/assets/d4dfafc1-8189-4d29-8e31-f8b46239274f)


### Zip-file with the report & the excel-xlsx
[commercial_table_with_no_dataset_TG-BIRT-4.17.zip](https://github.com/user-attachments/files/16716310/commercial_table_with_no_dataset_TG-BIRT-4.17.zip)

**Comment by speckyspooky** [comment]
Special information for you, I added an additional condition to the old excel-prototype to check if the diagonal-style-configuration is set.
The default is "none" (= style "unset"). So with this condition the diagonal will be avoid for the old excel-prototype too.
This fix will be available wth BIRT 4.17.

I would prefer to switch to the spudsoft emitter because this will be the future.

**Reply by PenghaiZhang** [reply]
Thank you @speckyspooky we are updating our code to use the spudsoft emitter. And look forward to v4.17.

**Comment by PenghaiZhang** [comment]
Hi @speckyspooky , the report designer in my team asked if the community has any guidance or documentation about the best practices of designing reports ? They'd like to update their knowledge in order to avoid the grid/table issue you identified in our testing report.
